pyvista/examples/_dataset_loader.py

```diff
@@ -81,8 +81,9 @@ def num_files(self) -> int:
 
         If a path is a folder, the number of files contained in the folder is returned.
         """
-        path = [path] if isinstance(path := self.path, str) else path
-        return sum([1 if os.path.isfile(p) else len(_get_all_nested_filepaths(p)) for p in path])
+        path = self.path
+        paths = [path] if isinstance(path, str) else path
+        return sum([1 if os.path.isfile(p) else len(_get_all_nested_filepaths(p)) for p in paths])
```

Comment:
Unrelated, but we can use a generator comprehension here
```suggestion
        return sum(1 if os.path.isfile(p) else len(_get_all_nested_filepaths(p)) for p in paths)
```